internal/blockchain/v0/reactor.go

```diff
@@ -560,6 +563,10 @@ FOR_LOOP:
 					panic(fmt.Sprintf("failed to process committed block (%d:%X): %v", first.Height, first.Hash(), err))
 				}
 
+				if r.consReactor != nil {
```

Comment:
This seems like a code smell to me. We should either be emitting metrics here directly or in consensus directly.

Reply:
The v0/reactor doesn't have the metrics, it only has a `consensusReactor` interface.
And I believe the fast-sync mode doesn't run the `consensus.state`, that's why I add a method in the `consensus/Reactor`

Reply:
redirected to use consensus/state and make `recordMetrics` accessible.